packages/serverless/src/backend_connector.js

```diff
@@ -156,9 +167,9 @@ function send(resourcePath, payload, destroySocketAfterwards, callback) {
     rejectUnauthorized: !disableCaCheck
   };
   if (!legacyTimeoutHandling) {
-    options.timeout = backendTimeout;
+    options.timeout = layerExtensionHasFailed ? backendTimeout : layerExtensionTimeout;
   }
-  if (proxyAgent) {
+  if (proxyAgent && layerExtensionHasFailed) {
```

Comment:
Does the extension already respect the proxy settings?

Reply:
Yes!